core/src/main/java/io/jenkins/pluginhealth/scoring/probes/ThirdPartyRepositoryDetectionProbe.java

```diff
@@ -51,7 +53,8 @@ protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
             return ProbeResult.error(KEY, e.getMessage());
 
         }
-        return ProbeResult.success(KEY, "The plugin has no third party repositories");
+        return allRepositories.size()>0 ? ProbeResult.success(KEY, "The plugin has no third party repositories")
+            : ProbeResult.failure(KEY, "No repositories detected");
```

Comment:
"No repositories detected" message seems more apt than "No third party repositories detected" for the case when the pom file contains no `<repository>`.

For this I would need to write a new test because the parameterized test won't work here? Shall I write a new test case?

What do you think?

Reply:
> "No repositories detected" message seems more apt than "No third party repositories detected" for the case when the pom file contains no <repository>.

good idea.

> For this I would need to write a new test because the parameterized test won't work here? Shall I write a new test case?

what is the easier: new parameter in the test method or a new test? This is entirely up to you.